src/main/java/frc/robot/subsystems/drivetrain/SwerveModule.java

```diff
@@ -0,0 +1,186 @@
+package frc.robot.subsystems.drivetrain;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+import edu.wpi.first.wpilibj.controller.LinearQuadraticRegulator;
+import edu.wpi.first.wpilibj.estimator.KalmanFilter;
+import edu.wpi.first.wpilibj.system.LinearSystem;
+import edu.wpi.first.wpilibj.system.LinearSystemLoop;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpiutil.math.Nat;
+import edu.wpi.first.wpiutil.math.VecBuilder;
+import edu.wpi.first.wpiutil.math.numbers.N1;
+import frc.robot.Constants;
+import frc.robot.subsystems.UnitModel;
+import frc.robot.utils.Utils;
+
+import static frc.robot.Constants.NOMINAL_VOLTAGE;
+import static frc.robot.Constants.SwerveDrive.TICKS_PER_DEGREE_ANGLE;
+import static frc.robot.Constants.SwerveDrive.TICKS_PER_METER_DRIVE;
+import static frc.robot.Constants.SwerveModule.*;
+
+public class SwerveModule extends SubsystemBase {
+    private final UnitModel angle_unitModel = new UnitModel(TICKS_PER_DEGREE_ANGLE);
+    private final UnitModel drive_unitModel = new UnitModel(TICKS_PER_METER_DRIVE);
+    private final LinearSystemLoop<N1, N1, N1> linearSystemLoop;
+    private WPI_TalonFX driveMotor;
+    private WPI_TalonSRX angleMotor;
```

Comment:
"steering" sounds better to me, this is also the term used in Ether's paper on Chief Delphi.
Make sure you rename it throughout the whole project.